core/src/test/java/de/jplag/reporting/reportobject/ReportObjectFactoryTest.java

```diff
@@ -1,14 +1,29 @@
 package de.jplag.reporting.reportobject;
 
+import java.io.File;
+
 import org.junit.jupiter.api.Assertions;
 import org.junit.jupiter.api.Test;
 
+import de.jplag.JPlagResult;
+import de.jplag.TestBase;
+import de.jplag.exceptions.ExitException;
 import de.jplag.reporting.reportobject.model.Version;
 
-class ReportObjectFactoryTest {
+class ReportObjectFactoryTest extends TestBase {
     @Test
     void testVersionLoading() {
         Assertions.assertNotNull(ReportObjectFactory.REPORT_VIEWER_VERSION);
         Assertions.assertNotEquals(Version.DEVELOPMENT, ReportObjectFactory.REPORT_VIEWER_VERSION);
     }
+
+    @Test
+    void testCreateAndSaveReportWithBasecode() throws ExitException {
+        JPlagResult result = runJPlag("basecode", it -> it.withBaseCodeSubmissionDirectory(new File(BASE_PATH, "basecode-base")));
+        String path = "JPlag\\core\\src\\test\\resources\\de\\jplag\\samples\\output\\submissions";
```

Comment:
Please reuse the base path of the test base and avoid system-specific separators with `Path.of`.